xdsl/dialects/riscv.py

```diff
@@ -346,13 +351,55 @@ def parse(cls, parser: Parser) -> Self:
             regions=regions,
         )
 
+    @classmethod
+    def parse_args(cls, parser: Parser) -> Sequence[UnresolvedOperand]:
+        if arg := parser.parse_optional_unresolved_operand():
```

Comment:
I think @math-fehr added an optional list without delimiter helper recently, could you please take a look?

Reply:
It seems there are some issues with `parser.parse_optional_undelimited_comma_separated_list`.

Reply:
This function should be put directly in the `parser` I would argue.
Also, what issue do you have with this function? `parser.parse_optional_undelimited_comma_separated_list(parser.parse_optional_unresolved_operand, parser.parse_unresolved_operand)` should work without problems.

Reply:
In the new custom syntax, both `args` and attributes are comma separated.

`riscv.op %0, %1, 123, "comment" (regions) ...`